pgjdbc/src/main/java/org/postgresql/ssl/LazyKeyManager.java

```diff
@@ -160,9 +159,19 @@ public String chooseServerAlias(String keyType, Principal[] issuers, Socket sock
     return (alias == null ? new String[]{} : new String[]{alias});
   }
 
+  private static byte[] readFileFully(String path) throws IOException {
+    RandomAccessFile raf = new RandomAccessFile(path, "r");
+    try {
+      byte[] ret = new byte[(int) raf.length()];
+      raf.readFully(ret);
+      return ret;
+    } finally {
+      raf.close();
```

Comment:
ideally a possible IOException should be caught here

Reply:
and log it ?

Reply:
Trapping IOExceptions on close is an anti-pattern. If there's truly an issue closing the resource then you'd want that to bubble up as well.

For files IO specifically I don't think they ever throw anyway but even if they did you'd want to catch it upstream in the catch block of the caller (which then wraps it in a PSQLException).